tests/rptest/scale_tests/upgrade_availability_test.py

```diff
@@ -0,0 +1,243 @@
+            replicas_status = []
+
+            for host in hosts:
+                res = self.admin._get_configuration(host,
+                                                    namespace='kafka',
+                                                    topic=self.topic,
+                                                    partition=0)
+
+                if 'status' not in res:
+                    return False
+                else:
+                    status = res['status']
+                    self.logger.debug(f'Movement status: {status}')
+                    replicas_status.append(status == 'in_progress')
+
+            return all(replicas_status)
+
+        wait_until(check_status_on_all_replicas, timeout_sec=120)
+
+    def restart_node_and_wait_partition_movement(self, node_id: int):
+        self.logger.info(f'Restarting node {node_id} with version 22.1.3')
+        node_to_upgrade = self.redpanda.get_node(node_id)
+        self.restart_node_w_version(node_to_upgrade, '22.1.3-1')
+
+        # Make sure partition movement is still in progress
+        self.confirm_in_progress_partition_movement(hosts=self.rp_hostnames)
+
+        # Make sure the producer is still writing
+        records_written_after_upgrade = self.producer.record_count
+        wait_until(
+            lambda: self.producer.record_count > records_written_after_upgrade,
+            timeout_sec=60,
+            err_msg='Producer stopped writing')
+
+        # Wait for partition movement to complete
+        self.admin.wait_stable_configuration(topic=self.topic,
+                                             namespace='kafka',
+                                             replication=3,
+                                             timeout_s=120)
+
+        # Terminate when the producer finishes
+        self.producer.wait(timeout_sec=1200)
+
+    @cluster(num_nodes=5, log_allow_list=CHAOS_LOG_ALLOW_LIST)
+    def test_node_will_not_be_in_replica_set(self):
+        rs_tuple = self.run_workload_and_get_replica_sets()
+        nodes_in_replica_set = rs_tuple[0]
+        nodes_not_in_replica_set = rs_tuple[1]
+
+        # Execute partition movement such that node X will no longer be in the
+        # replica set. Therefore, move the partition to a node that is not in
+        # the replica set.
+        node_x = choice(nodes_in_replica_set)
+        node_y = choice(nodes_not_in_replica_set)
+        print(f'Move partition from {node_x} to {node_y}')
+        part_movement_th = threading.Thread(
+            target=self.move_from_src_to_dest,
+            args=(node_x, node_y),
+        )
+        part_movement_th.start()
+
+        # Check that partition movement started/is on-going
+        self.confirm_in_progress_partition_movement(hosts=self.rp_hostnames)
+
+        # After node restart, probably within the below function,
+        # again check that parttion movement started/is on-going
+        self.restart_node_and_wait_partition_movement(node_id=node_x)
```

Comment:
how long does the partition movement last? do we have time to let it a bit before restarting, it might be beneficial from reproducing perspective if we let it run a bit

Reply:
I'll re-run the test because I don't recall how long movement lasts.
I'll also look into letting movement run for a few seconds after we initiate node upgrade.

Reply:
A little bit of timeout between starting movement and node upgrade.

Reply:
done but I did not get a change to test if the amount of time I'm waiting for is sufficient.